### PR TITLE
Fix: Empty <amp-experiment> blocks boilerplate removal

### LINKED ISSUE #578: Empty <amp-experiment> blocks boilerplate removal
The check for `<amp-experiment>` in the `ServerSideRendering` transformer is missing a check to make sure the `<amp-experiment>` is not just empty.
https://github.com/ampproject/amp-toolbox/blob/2a81fc3795a0a5354920f7c2855af9900eecacb6/packages/optimizer/lib/transformers/ServerSideRendering.js#L88

In the Go implementation, you can see an additional `isAmpExperimentUsed()` check: https://github.com/ampproject/amppackager/blob/v2/transformer/transformers/serversiderendering.go#L127

I'm currently facing the issues that the spec files seem to be wrong as this check is missing, so my tests are failing here:
https://travis-ci.org/ampproject/amp-wp/jobs/640983826#L1270-L1340

### PR DESCRIPTION
**Pull request by sebastianbenz**
Fixes #578

### CODE REVIEW

**Comment by sebastianbenz** [comment]
@schlessera PTAL

File: packages/optimizer/lib/transformers/ServerSideRendering.js [hunk 1]
```diff
@@ -85,7 +85,7 @@ class ServerSideRendering {
       // amp-experiment is a render delaying extension iff the tag is used in
       // the doc. We check for that here rather than checking for the existence
       // of the amp-experiment script in IsRenderDelayingExtension below.
-      if (node.tagName === 'amp-experiment') {
+      if (node.tagName === 'amp-experiment' && !this.isAmpExperimentUsed(node)) {
```
Comment (schlessera) [inline packages/optimizer/lib/transformers/ServerSideRendering.js]: The logic is the wrong way around:
```suggestion
      if (node.tagName === 'amp-experiment' && this.isAmpExperimentUsed(node)) {
```

**Comment by sebastianbenz** [comment]
Hah! Fooled myself there. Thanks for noticing. Should be fixed now. I've also added more tests and improved the test names. @schlessera PTAL

File: packages/optimizer/lib/transformers/ServerSideRendering.js (hunk 1 above)
Comment (sebastianbenz) [reply packages/optimizer/lib/transformers/ServerSideRendering.js]: done

**Comment by schlessera** [comment]
LGTM!

**Comment by sebastianbenz** [comment]
Thanks for the review!

**Pull request merged**